apps/web/src/components/settings/RegistrationForm/RegisterFormSettings.tsx

```diff
@@ -92,11 +92,10 @@ export default function RegisterFormSettings({
 	const { isSubmitSuccessful, isSubmitted, errors } = form.formState;
 	const hasErrors = !isSubmitSuccessful && isSubmitted;
 	const [uploadedFile, setUploadedFile] = useState<File | null>(null);
-	const resumeLink: string = data.resume ?? c.noResumeProvidedURL;
-	// @ts-ignore
-	let f = new File([data.resume], resumeLink.split("/").pop());
+	let oldResumeLink: string = data.resume ?? c.noResumeProvidedURL;
+	let f = new File([data.resume], oldResumeLink.split("/").pop()!);
 	useEffect(() => {
-		if (resumeLink === c.noResumeProvidedURL) setUploadedFile(null);
+		if (oldResumeLink === c.noResumeProvidedURL) setUploadedFile(null);
```

Comment:
Is the use of a `useEffect` needed here? Without any dependency arguments, this will only run once which would be the same thing as just having it inline with an assignment statement.

Reply:
Without the `useEffect` it will infinite-rerender